project_team_leave_management/models/supplier_leaves.py

```diff
@@ -0,0 +1,96 @@
+# See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class SupplierLeaves(models.Model):
+    _name = "supplier.leaves"
+    _description = "Supplier Leaves"
+    _rec_name = "partner_id"
+
+    @api.depends("date_from", "date_to")
+    def _compute_leave_duration(self):
+        """
+        Compute the days Duration.
+        """
+        leave_duration = 0
+        for leaves in self:
+            if leaves.date_from and leaves.date_to:
+                leave_duration = 1 + (leaves.date_to - leaves.date_from).days
+            leaves.leave_duration = leave_duration
+
+    def _get_partner_domain(self):
+        """
+        Get the Partner domain to limit only
+        to a supplier compnay's resources.
+        """
+        domain = []
+        if self.env.user.supplier_company_id:
+            domain.append(("id", "in", self.env.user.supplier_company_id.child_ids.ids))
+        return domain
```

Comment:
Change this function's body to:
<pre> return [("id", "in", self.env.user.supplier_company_id.child_ids.ids)] if self.env.user.supplier_company_id else [] </pre>